mypy/build.py

```diff
@@ -809,6 +813,43 @@ def remove_cwd_prefix_from_path(fscache: FileSystemCache, p: str) -> str:
     return p
 
 
+SITE_PACKAGES = '''
+from __future__ import print_function
+from distutils.sysconfig import get_python_lib
+import site
+if hasattr(site, 'getusersitepackages') and hasattr(site, 'getsitepackages'):
+    user_dir = site.getusersitepackages()
+    print(repr(site.getsitepackages() + [user_dir]))
+else:
+    print(repr([get_python_lib()]))
+'''
```

Comment:
How about actually reusing the code path?
```python
code = '''
from __future__ import print_function

def main():
	from distutils.sysconfig import get_python_lib
	import site
	if hasattr(site, 'getusersitepackages') and hasattr(site, 'getsitepackages'):
	    user_dir = site.getusersitepackages()
	    return site.getsitepackages() + [user_dir]
	else:
	    return [get_python_lib()]

if __name__ == '__main__':
	print(repr(main()))
'''

if active_python:
    exec_globals = {}
    exec(code, exec_globals)
    res = exec_globals['main']()
else:
    res = ast.literal_eval(call(['python', '-c', code]))
```

Reply:
That seems rather hacky and slower for little reason.

Reply:
A slightly less hacky/slow version would be to put this in it's own file, and do:
```
from . import the_file
if this_python:
    res = the_file.main()
else:
    res = ast.literal_eval(call(['python', '-c', the_file.__file__]))
```

with the rationale of removing code duplication.

Reply:
I'm not convinced forcing them to be the same makes sense. One is printing the repr of objects, the other is returning the objects. I don't like the idea of coercing code to fit to one or the other just for the sake of symmetry.

Reply:
Both are trying to construct exactly the same objects, all that changes is how they return them. One can just return it, the other goes via `repr`/`literal_eval`

Reply:
Bear in mind that having `this_python` as a code path at all is just an optimization of the `else` case, so it definitely ought to be symmetric.

Reply:
I suppose I find the second acceptable. Why would writing bytecode be dangerous? It is versioned for a reason.